airflow/models.py

```diff
@@ -1574,6 +1574,8 @@ def dry_run(self):
     def handle_failure(self, error, test_mode=False, context=None, session=None):
         self.log.exception(error)
         task = self.task
+        session = settings.Session()
+        self.exception = error
```

Comment:
Adding a property to the TaskInstance object feels the wrong place for this - the context object feels like a more appropriate place.